tests/test_transforms/test_encoders/test_mean_encoder_transform.py

```diff
@@ -0,0 +1,250 @@
+import numpy as np
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+
+from etna.datasets import TSDataset
+from etna.datasets import generate_ar_df
+from etna.metrics import MSE
+from etna.models import LinearMultiSegmentModel
+from etna.pipeline import Pipeline
+from etna.transforms import FilterFeaturesTransform
+from etna.transforms import MeanEncoderTransform
+from tests.test_transforms.utils import assert_transformation_equals_loaded_original
+from tests.utils import select_segments_subset
+
+
+@pytest.fixture
+def category_ts() -> TSDataset:
+    df = generate_ar_df(start_time="2001-01-01", periods=6, n_segments=2)
+    df["target"] = [1, 2, 8, 4, np.NaN, 5] + [6, 7, 8, 9, 10, 11]
+
+    df_exog = generate_ar_df(start_time="2001-01-01", periods=8, n_segments=2)
+    df_exog.rename(columns={"target": "regressor"}, inplace=True)
+    df_exog["regressor"] = ["A", "B", np.NaN, "A", np.NaN, "B", "C", "A"] + ["A", "B", "A", "A", "A", np.NaN, "A", "C"]
```

Comment:
What if there are NaNs in the beginning of the segments?

Reply:
NaN in target or in category